tests/rados/test_four_node_ecpool.py

```diff
@@ -103,245 +122,267 @@ def run(ceph_cluster, **kw):
-        )
-        if not rados_obj.check_inactive_pgs_on_pool(pool_name=pool_name):
-            log.error(f"Inactive PGs found on pool : {pool_name}")
-            inactive_pgs += 1
-
-        # Adding the removed OSD back and checking the cluster status
-        log.debug("Adding the removed OSD back and checking the cluster status")
-        utils.add_osd(ceph_cluster, host.hostname, dev_path, target_osd)
-        method_should_succeed(wait_for_device_rados, host, target_osd, action="add")
-        time.sleep(30)
-        log.debug(
-            "Completed addition of OSD post removal. Checking for inactive PGs post OSD addition"
-        )
-        if not rados_obj.check_inactive_pgs_on_pool(pool_name=pool_name):
-            log.error(f"Inactive PGs found on pool : {pool_name}")
-            inactive_pgs += 1
+            if not rados_obj.check_inactive_pgs_on_pool(pool_name=pool_name):
+                log.error(f"Inactive PGs found on pool : {pool_name}")
+                inactive_pgs += 1
+
+            # Adding the removed OSD back and checking the cluster status
+            log.debug("Adding the removed OSD back and checking the cluster status")
+            utils.add_osd(ceph_cluster, host.hostname, dev_path, target_osd)
+            method_should_succeed(wait_for_device_rados, host, target_osd, action="add")
+            time.sleep(30)
+            log.debug(
+                "Completed addition of OSD post removal. Checking for inactive PGs post OSD addition"
+            )
+            if not rados_obj.check_inactive_pgs_on_pool(pool_name=pool_name):
+                log.error(f"Inactive PGs found on pool : {pool_name}")
+                inactive_pgs += 1
 
-        if inactive_pgs > 10:
-            log.error("Found inactive PGs on the cluster during OSD removal/Addition")
-            raise Exception("Inactive PGs during stop error")
+            if inactive_pgs > 10:
+                log.error(
+                    "Found inactive PGs on the cluster during OSD removal/Addition"
+                )
+                raise Exception("Inactive PGs during stop error")
 
-        # Checking cluster health after OSD removal
-        method_should_succeed(rados_obj.run_pool_sanity_check)
-        log.info(
-            f"Addition of OSD : {target_osd} back into the cluster was successful, and the health is good!"
-        )
+            # Checking cluster health after OSD removal
+            method_should_succeed(rados_obj.run_pool_sanity_check)
+            log.info(
+                f"Addition of OSD : {target_osd} back into the cluster was successful, and the health is good!"
+            )
 
-        utils.set_osd_devices_unmanaged(ceph_cluster, target_osd, unmanaged=False)
-        log.info("Completed the removal and addition of OSD daemons")
+            utils.set_osd_devices_unmanaged(ceph_cluster, target_osd, unmanaged=False)
+            log.info("Completed the removal and addition of OSD daemons")
```

Comment:
Looks good for now. line no : 387, begins with host removal + addition. That can be fine tuned and added as scenario later.